builtin/providers/rancher/resource_rancher_registration_token.go

```diff
@@ -0,0 +1,181 @@
+package rancher
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	rancherClient "github.com/rancher/go-rancher/client"
+)
+
+func resourceRancherRegistrationToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceRancherRegistrationTokenCreate,
+		Read:   resourceRancherRegistrationTokenRead,
+		Delete: resourceRancherRegistrationTokenDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
We should use `Optinal: true` here so that we can terraformize what exists without having to update to token name in the database using SQL queries because this attribute is not updatable using the API.